packages/vx-demo/src/docs-v2/examples/vx-shape-pie/Example.tsx

```diff
@@ -0,0 +1,216 @@
+      <Group top={centerY - margin.top} left={centerX}>
+        <Pie
+          data={
+            selectedBrowser ? browsers.filter(({ label }) => label === selectedBrowser) : browsers
+          }
+          pieValue={usage}
+          outerRadius={radius - 80}
+          innerRadius={radius - 120}
+          cornerRadius={3}
+          padAngle={0.005}
+        >
+          {pie => (
+            <AnimatedPie<BrowserUsage>
+              {...pie}
+              animate={animate}
+              getKey={arc => arc.data.label}
+              onClickDatum={({ data: { label } }) =>
+                animate &&
+                setSelectedBrowser(selectedBrowser && selectedBrowser === label ? null : label)
+              }
+              getColor={arc => getBrowserColor(arc.data.label)}
+            />
+          )}
+        </Pie>
+        <Pie
+          data={
+            selectedAlphabetLetter
+              ? letters.filter(({ letter }) => letter === selectedAlphabetLetter)
+              : letters
+          }
+          pieValue={frequency}
+          pieSortValues={() => -1}
+          outerRadius={radius - 135}
+        >
+          {pie => (
+            <AnimatedPie<LetterFrequency>
+              {...pie}
+              animate={animate}
+              getKey={({ data: { letter } }) => letter}
+              onClickDatum={({ data: { letter } }) =>
+                animate &&
+                setSelectedAlphabetLetter(
+                  selectedAlphabetLetter && selectedAlphabetLetter === letter ? null : letter,
+                )
+              }
+              getColor={({ data: { letter } }) => getLetterFrequencyColor(letter)}
+            />
+          )}
+        </Pie>
+      </Group>
+      {animate && (
+        <text
+          textAnchor="end"
+          x={width - 16}
+          y={height - 16}
+          fill="white"
+          fontSize={11}
+          fontWeight={300}
+          pointerEvents="none"
+        >
+          Click segments to update
+        </text>
+      )}
+    </svg>
+  );
+}
+
+// react-spring transition definitions
+type AnimatedStyles = { startAngle: number; endAngle: number; opacity: number };
+
+const fromLeaveTransition = ({ endAngle }: PieArcDatum<any>) => ({
+  // enter from 360° if end angle is > 180°
+  startAngle: endAngle > Math.PI ? 2 * Math.PI : 0,
+  endAngle: endAngle > Math.PI ? 2 * Math.PI : 0,
+  opacity: 0,
+});
+const enterUpdateTransition = ({ startAngle, endAngle }: PieArcDatum<any>) => ({
+  startAngle,
+  endAngle,
+  opacity: 1,
+});
+
+type AnimatedPieProps<Datum> = ProvidedProps<Datum> & {
+  animate?: boolean;
+  getKey: (d: PieArcDatum<Datum>) => string;
+  getColor: (d: PieArcDatum<Datum>) => string;
+  onClickDatum: (d: PieArcDatum<Datum>) => void;
+  delay?: number;
+};
+
+function AnimatedPie<Datum>({
+  animate,
+  arcs,
+  path,
+  getKey,
+  getColor,
+  onClickDatum,
+}: AnimatedPieProps<Datum>) {
+  // @ts-ignore react-spring doesn't like this overload
```

Comment:
spent a bunch of time on this, should be supported by the docs and it works, but TS didn't like it